src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -60,11 +66,23 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
       String userId = userService.getCurrentUser().getUserId();
       Key userKey = KeyFactory.createKey("User", userId);
-      try {
-        datastore.get(userKey);
-        // TODO(#16): Use query instead of get.
-      } catch (Exception e) {
-        datastore.put(new Entity("User", userId));
+      Filter keyFilter =
+          new FilterPredicate(Entity.KEY_RESERVED_PROPERTY, FilterOperator.EQUAL, userKey);
+      Query userQuery = new Query("User").setFilter(keyFilter);
+      PreparedQuery checkForUser = datastore.prepare(userQuery);
+
+      // If user entity doesn't already exist, create one.
+      if (checkForUser.countEntities(FetchOptions.Builder.withLimit(10)) == 0) {
```

Comment:
Why is the limit equals 10? Could you please add a constant?

Reply:
Done